### PR TITLE
Implement selectByOccurrences on primitive Bags.

### PR DESCRIPTION
**Pull request by donraab**
Signed-off-by: Donald Raab <Donald.Raab@gs.com>

### CODE REVIEW

File: eclipse-collections-code-generator/src/main/resources/impl/bag/mutable/primitiveHashBag.stg [hunk 1]
```diff
@@ -194,6 +195,19 @@ public final class <name>HashBag extends Abstract<name>Iterable implements Mutab
         this.items.forEachKeyValue(procedure);
     }
 
+    public <name>HashBag selectByOccurrences(IntPredicate predicate)
```
Comment (nikhilnanivadekar) [inline eclipse-collections-code-generator/src/main/resources/impl/bag/mutable/primitiveHashBag.stg]: The return type should be Mutable<name>Bag.

Comment (itohro) [inline eclipse-collections-code-generator/src/main/resources/impl/bag/mutable/primitiveHashBag.stg]: I think this is fine. We usually use more concrete return types for select*() methods in concrete classes (e.g. `UnifiedSet#select()` returns `UnifiedSet`, `FastList#select()` returns `FastList` etc.). 

Think @motlin can confirm the background on this convention.

**Comment by itohro** [comment]
Looks good to me overall. 

Please add the change to `RELEASE_NOTE_DRAFT.md` then we could merge this PR.

**Comment by itohro** [comment]
It seems there is a checkstyle failure. Please fix that.

04:39:35:262 [ERROR] src/main/java/org/eclipse/collections/impl/bag/mutable/primitive/BooleanHashBag.java:[522,50] (whitespace) ParenPad: ')' is preceded with whitespace.
04:39:35:262 [ERROR] src/main/java/org/eclipse/collections/impl/bag/mutable/primitive/BooleanHashBag.java:[552,50] (whitespace) ParenPad: ')' is preceded with whitespace.

**Comment by donraab** [comment]
Fixed the checkstyle issues.

**Pull request merged**